sklearn/model_selection/_search.py

```diff
@@ -105,12 +106,28 @@ def __init__(self, param_grid):
         # check if all entries are dictionaries of lists
         for grid in param_grid:
             if not isinstance(grid, dict):
-                raise TypeError("Parameter grid is not a dict ({!r})".format(grid))
-            for key in grid:
-                if not isinstance(grid[key], Iterable):
+                raise TypeError(
+                    f"Parameter grid for parameter '{grid}' is not a dict ({grid})"
```

Comment:
The grid is not a parameter, but the parameter grid itself. For now, we can be consistent with:

https://github.com/scikit-learn/scikit-learn/blob/c29fe81218833f8f14b28cf6d5374201ac8f3c13/sklearn/model_selection/_search.py#L259-L261

and use the same string format:

```suggestion
                    f"Parameter grid is not a dict ({grid!r})"
```